src/src/com/microsoft/aad/adal/AuthenticationParameters.java

```diff
@@ -166,17 +167,17 @@ public static void createFromResourceUrl(Context context, final URL resourceUrl,
             public void run() {
                 HashMap<String, String> headers = new HashMap<String, String>();
                 headers.put(WebRequestHandler.HEADER_ACCEPT, WebRequestHandler.HEADER_ACCEPT_JSON);
-                HttpWebResponse webResponse = sWebRequest.sendGet(resourceUrl, headers);
 
-                if (webResponse != null) {
-                    if(webResponse.getResponseException() != null) {
-                        onCompleted(webResponse.getResponseException(), null);
-                    }
+                HttpWebResponse webResponse = null;
+                try {
+                    webResponse = sWebRequest.sendGet(resourceUrl, headers);
                     try {
                         onCompleted(null, parseResponse(webResponse));
                     } catch (ResourceAuthenticationChallengeException exc) {
```

Comment:
nit: final
Sorry, I always insist on making variables final if it's not going to be changed.